### ISSUE TITLE
Custom items aren't assigned any category (and therefore don't show up in the main window)

### ISSUE THREAD

**Issue by Duckwhale** [WowRarity/Rarity#323]
In reworking the validation logic, I detected some issues (shocking, I know). One of them is this:

* Custom items don't have a category set (expansion)

This means they aren't ever displayed in the tooltip (as far as I can tell). It also means the new validation routine isn't having any of that.

Possible solutions:

* Make the field optional in the validation logic
* Force users to assign a category OR automatically put it in a new undefined category, that would also be displayed

Personally, I'd bonk people over the head when they try to enter a new item. But obviously you can't do that retroactively, and I'd assume lots of people added lots of items without categories, and no one seems to have noticed they didn't show up in the tooltip itself.

So I guess we should add a new group for the items that aren't assigned any one, just pretending it's a "zero-th" expansion (``NONE``?) maybe, and also add a dropdown in the Options UI to have people select a different one. Hopefully, they'll notice it says NONE and change it. Then we can make the field required again.

For now I'll make it optional, but it doesn't seem right.

**Comment by godejord** [comment]
> This means they aren't ever displayed in the tooltip (as far as I can tell). It also means the new validation routine isn't having any of that.

Just wanted to chime in and say that they do actually appear in the tooltip even without any category. I've been using custom items to track some apperances, and they show up just fine.

**Comment by Duckwhale** [comment]
You're right, there's actually a group called ``Custom``. No idea why, but for some reason I didn't see it. 

However, the validation still fails because the field ``cat`` is not set:

![image](https://user-images.githubusercontent.com/16489133/113990682-009d9680-9852-11eb-8f2f-7995084f025e.png)

This is using the "semi-official" illusions item set linked on the WowAce page.
